### PR TITLE
Developments on checks and tests. Added to README and new roadmap.md file 

### PR DESCRIPTION
**Pull request by daltonkell**
@brianmckenna @Bobfrat

### CODE REVIEW

**Comment by daltonkell** [comment]
@Bobfrat @brianmckenna All Travis CI tests passed

File: .gitignore [hunk 1]
```diff
@@ -60,6 +60,3 @@ docs/_build/
 
 # PyBuilder
 target/
-
-
-*.nc
```
Comment (brianmckenna) [inline .gitignore]: we don't want to exclude .nc files, they're used in tests right?

File: .pytest_cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "cc_plugin_ugrid/checker_10_test.py::TestUgridChecker10::test_check_location_loop": true
```
Comment (brianmckenna) [inline .pytest_cache/v/cache/lastfailed]: we DO want to add .pytest_cache to .gitignore

File: pytest.ini [hunk 3]
```diff
@@ -1,3 +1,5 @@
 [pytest]
 norecursedirs = requirements
 python_functions = *_test
+log_cli=true
```
Comment (brianmckenna) [inline pytest.ini]: pull these `log_*` out, users can add in as needed

File: cc_plugin_ugrid/resources/fvcom_reference.cdl [hunk 4]
```diff
@@ -0,0 +1,110 @@
+netcdf fvcom_reference {
```
Comment (brianmckenna) [inline cc_plugin_ugrid/resources/fvcom_reference.cdl]: change to match the associated .nc file:

`netcdf fvcom {`

and update .cdl filename, better to have a clear map between what is responsible for .nc

File: cc_plugin_ugrid/resources/adcirc_reference.cdl [hunk 5]
```diff
@@ -0,0 +1,53 @@
+netcdf adcirc_reference {
```
Comment (brianmckenna) [inline cc_plugin_ugrid/resources/adcirc_reference.cdl]: change to match the associated .nc file:

`netcdf adcirc {`

and update .cdl filename, better to have a clear map between what is responsible for .nc

is this .cdl the dump of existing adcirc file?

File: cc_plugin_ugrid/checker_10.py [hunk 6]
```diff
@@ -166,5 +215,199 @@ def check_connectivity(self, ds):
...
+        desc = '"location" & "mesh" strongly encouraged for variables with' + \
+                ' dimensions "time" , "node", or "faces" (elements)'
+
+        # check attributes
+        for var in ds.variables:
+            dims = ds[var].dimensions
+            m = ''
+            try:
+                assert any(x in dims for x in [self.time_dim, self.node_dim, self.face_dim])
+                # logger.debug('{} -- {} -- {}'.format(self.time_dim, self.node_dim, self.face_dim))
```
Comment (brianmckenna) [inline cc_plugin_ugrid/checker_10.py]: cleanup: aisle 395

File: cc_plugin_ugrid/checker_10.py [hunk 7]
```diff
@@ -120,21 +121,69 @@ def check_node_coordinates_exist(self, ds):
             return None
         except AttributeError:
             # No node_dimensions attribute... there are larger issues
+            self.node_dim = None
```
Comment (brianmckenna) [inline cc_plugin_ugrid/checker_10.py]: is it possible to move 

`self.node_dim = None` above, so it's set to None, only to be replaced by final condition?

File: cc_plugin_ugrid/checker_10.py [hunk 8]
```diff
@@ -120,21 +121,69 @@ def check_node_coordinates_exist(self, ds):
...
             return None
         except AssertionError:
             m = '"node_coordinates" member "{}" is not a variable'.format(n)
             messages.append(m)
+            self.node_dim = None
+        else:
+            # verify that the node coordinates' dimensions are equivalent
+            node_coord = n
```
Comment (brianmckenna) [inline cc_plugin_ugrid/checker_10.py]: what is `n` here? Is it the last iteration of above loop?

File: cc_plugin_ugrid/resources/fvcom_reference.cdl (hunk 4 above)
Comment (daltonkell) [reply cc_plugin_ugrid/resources/fvcom_reference.cdl]: @brianmckenna So which name do you want me to keep?

File: cc_plugin_ugrid/checker_10.py [hunk 9]
```diff
@@ -120,21 +121,69 @@ def check_node_coordinates_exist(self, ds):
...
+            try:
+                for ncoord in ncs:
+                    assert ds[ncoord].dimensions == ds[node_coord].dimensions
+            except AssertionError:
+                # names are not equivalent
+                m = 'names of node coordinates appear to be different'
```
Comment (brianmckenna) [inline cc_plugin_ugrid/checker_10.py]: I feel like my response to this message would be "oh, do they now?", lets make an assertion:

node_coordinate values must match.....though I'm uncertain what we're matching.

File: cc_plugin_ugrid/checker_10.py [hunk 10]
```diff
@@ -166,5 +215,199 @@ def check_connectivity(self, ds):
...
+        """
+        Loops through each variable to see if its dimensions contain time,
+        nodes, or faces. If so, imlpements a check for a 'location'
+        attribute in the variable; fails if it does not exist.
+        """
+
+        # DEPENDENCIES
+        self.check_face_dimension(ds)
+        self.check_node_coordinates_exist(ds)
+        if ds.get_variables_by_attributes(standard_name='time'):
```
Comment (brianmckenna) [inline cc_plugin_ugrid/checker_10.py]: I don't think we need time here, there's no requirement a UGRID variable be time-varying. I may have led you down wrong path here. 🤕

File: cc_plugin_ugrid/checker_10.py (hunk 8 above)
Comment (daltonkell) [reply cc_plugin_ugrid/checker_10.py]: @brianmckenna Yes, as the last value of the node_coordinates variable should be the same dimensions. Does that make sense?

**Pull request merged**